### PR TITLE
Added support for the Appium Touch Actions

### PR DESCRIPTION
**Pull request by oecheverri**
Added support for: tap, longpress, swipe, longpressswipe to the You.i Engine Driver

### CODE REVIEW

File: lib/commands/gesture.js [hunk 1]
```diff
@@ -4,10 +4,24 @@ import { unwrapEl, youiEngineDriverReturnValues } from '../utils';
...
+
+commands.tapEl = async function (el, longPress) {
+
+  // perform a tap on the given element
+  // if longPress is true, the tap becomes a longPress action
+
   el = unwrapEl(el);
 
+  let commandName = 'tap';
+  if (longPress === true) {
```
Comment (jplandry) [inline lib/commands/gesture.js]: Is this necessary?

File: lib/commands/gesture.js [hunk 2]
```diff
@@ -25,8 +39,199 @@ commands.click = async function (el) {
...
+  if (elementId) {
+
+    // we are either tapping on the default location of the element
+    // top left has been specified of the element has been specified
+
+    if (x && y) {
+      await this.tapElXY(elementId, x, y, longPress);
+    } else {
+      // tapping with an element parameter is the same as click on a WebElement so they can
+      // filter through the same call stack
```
Comment (jplandry) [inline lib/commands/gesture.js]: `so they can fiilter through the same call stack` This doesn't add any information.

File: lib/commands/gesture.js [hunk 3]
```diff
@@ -25,8 +39,199 @@ commands.click = async function (el) {
...
+
+  }
+  if (moveTo.options.element) {
+    let {x, y} = await this.getLocation(moveTo.options.element);
+    let {width, height} = await this.getSize(moveTo.options.element);
+    endX += x || 0;
+    endY += y || 0;
+
+    // let's use the middle of the element to move to, unless an offset from
+    // top left has been specified of the element has been specified
```
Comment (jplandry) [inline lib/commands/gesture.js]: // let's use the middle of the element to move to, unless an offset from
 // **the** top left ~~has been specified~~ of the element has been specified

File: lib/commands/gesture.js [hunk 4]
```diff
@@ -25,8 +39,199 @@ commands.click = async function (el) {
...
+    let {width, height} = await this.getSize(press.options.element);
+
+
+    if (!press.options.x && !press.options.y && width && height) {
+      startX += width / 2;
+      startY += height / 2;
+    }
+
+    startX += x || 0;
+    startY += y || 0;
```
Comment (pfoster-youitv) [inline lib/commands/gesture.js]: This bit seems odd - might be good to have comment here.

The first part is checking if there is no press option x and y but there is a width and height. Calculating start X/Y by center of size.

Next part is updating that number based on the element location? If there is one?

File: lib/commands/gesture.js [hunk 5]
```diff
@@ -25,8 +39,199 @@ commands.click = async function (el) {
...
+    let {width, height} = await this.getSize(moveTo.options.element);
+    endX += x || 0;
+    endY += y || 0;
+
+    // let's use the middle of the element to move to, unless an offset from
+    // top left has been specified of the element has been specified
+    if (!press.options.x && !press.options.y && width && height) {
+      startX += width / 2;
+      startY += height / 2;
+    }
```
Comment (pfoster-youitv) [inline lib/commands/gesture.js]: Should this be startX/Y or endX/Y? Copy paste error?

Looks to be similar code to the above, just ordered differently and using endX/endY. Could it be made into a helper function that calculates control mid point or something? offset if any?

File: lib/commands/gesture.js (hunk 1 above)
Comment (pfoster-youitv) [inline lib/commands/gesture.js]: do we need the `=== true` part? (same for several below)

**Comment by pfoster-youitv** [comment]
A couple of minor comments, one suspected copy paste issue (?) but otherwise looks good. 👍

**Pull request merged**